packages/dataviews/src/field-types/utils/get-filter-by.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import type { Field, FilterByConfig, Operator } from '../../types';
+
+function getFilterBy< Item >(
+	field: Field< Item >,
+	defaultOperators: Operator[],
+	validOperators: Operator[]
+): Required< FilterByConfig > | false {
+	if ( field.filterBy === false ) {
+		return false;
+	}
+
+	if ( typeof field.filterBy === 'object' ) {
+		let operators = field.filterBy.operators;
+
+		// Assign default values if no operator was provided.
+		if ( ! operators || ! Array.isArray( operators ) ) {
+			operators = defaultOperators;
+		}
+
+		// Make sure only valid operators are included.
+		operators = operators.filter( ( operator ) =>
+			validOperators.includes( operator )
+		);
+
+		// If no operators are left at this point,
+		// the filters should be disabled.
+		if ( operators.length === 0 ) {
+			return false;
+		}
+
+		return {
+			isPrimary: !! field.filterBy.isPrimary,
+			operators,
+		};
+	}
+
+	if ( defaultOperators.length === 0 ) {
+		return false;
+	}
+
+	return {
+		isPrimary: false,
+		operators: defaultOperators,
+	};
```

Comment:
A few things seemed a bit strange in this function: filtering the default operators (unless we really want that extra safety net to prevent user error?); a seemingly superfluous `isArray` check (unless we believe consumers are not using TS?); the amount of branching and return statements; and the cumbersome requirement for consumers that they provide two identical arrays (`defaultOperators`, `validOperators`).

For the latter, maybe that explicitness is welcome, but otherwise I could be a nice touch to let `defaultOperators` default to `validOperators` when it is undefined.

For the rest, I found this refactor clearer, but this is probably very subjective:


```suggestion
	const operators =
		field.filterBy?.operators?.filter( ( operator ) =>
			validOperators.includes( operator )
		) ?? defaultOperators;

	// If no operators are left at this point, the filters should be disabled.
	if ( ! operators.length ) {
		return false;
	}

	return {
		isPrimary: !! field.filterBy?.isPrimary,
		operators,
	};
```

Reply:
I've addressed this at https://github.com/WordPress/gutenberg/pull/73546/commits/47f458cdd5e77541077fd7eff9b1ef4401548ef4